File: julia/hobbs4_2.jl

## using Distributions

# Function to calculate alpha and beta parameters for the beta distribution
function calculate_alpha_beta(μ, σ²)
    α = μ^2 - μ^3 - μ * σ² / σ²
    β = μ - 2μ^2 + μ^3 - σ² + μ * σ² / σ²
    return α, β
end

# Function to calculate probability density for a given observation y
function calculate_probability_density(y, α, β)
    dist = Beta(α, β)
    return pdf(dist, y)
end

# Given parameters
k = 0.01  # Decay rate
σ² = 6e-4  # Variance

# Calculate μt for t = 30
t = 30
μt = exp(-k * t)

# Calculate alpha and beta parameters
α30, β30 = calculate_alpha_beta(μt, σ²)

# Given observation
yt = 0.7

# Calculate probability density for yt at time t = 30
probability_density = calculate_probability_density(yt, α30, β30)

# Display results
println("Alpha for t = 30: ", α30)
println("Beta for t = 30: ", β30)
println("Probability density for yt = $yt at t = $t: ", probability_density)

File: R/hobbs4_2.R

# Example 4.2 Hobbs 2015: Probability nd beta Distribution

# Given the parameters
k <- 0.01
sigma2 <- 6e-4
t <- 30
observed_yt <- 0.7



# Function to calculate alpha and beta based on moment matching [[eq. in book]]
calculate_alpha_beta <- function(mu_t, sigma2) {
  alpha_t  <-  (mu_t^2 - mu_t^3 - (mu_t * sigma2)) / sigma2
  beta_t    <- (mu_t - 2 * mu_t^2 + mu_t^3 - sigma2 + (mu_t * sigma2)) / sigma2
  return(c(alpha_t, beta_t))
}



# Calculate alpha and beta for day 30
mu_t_30 <- exp(-k * t)
params_30 <- calculate_alpha_beta(mu_t_30, sigma2)
alpha_30 <- params_30[1]
beta_30 <- params_30[2]

# Calculate the probability density for yt = 0.7 at time t = 30
probability_density <- dbeta(observed_yt, alpha_30, beta_30)

# Display results
cat("Alpha for day",t,":", alpha_30, "\n")
cat("Beta for day",t,":", beta_30, "\n")
cat("Probability density for yt =",observed_yt," at time t =",t,":", probability_density, "\n")

